### ISSUE TITLE
Rename blacklist/whitelist to blocklist/allowlist

### ISSUE THREAD

**Issue by erichard** [0xERR0R/blocky#907]
A lots of projects have done the same including : google, angular, mysql, mozilla, etc

Beside the subject of offending terms, blocklist/allowlist is simpler to understand

**Comment by 0xERR0R** [comment]
We had already an attempt to do it: https://github.com/0xERR0R/blocky/pull/349. It is not just renaming, the terms are used in metrics, dashboards etc.

**Comment by Ornias1993** [comment]
> A lots of projects have done the same 

I don't think that's relevant. 
Even more so when it's either companies with shareholders or projects that are inherently political.

Not because of a personal preference, but because simply the fact that most companies/projects have a certain opinion, doesn't mean every one should.

> Beside the subject of offending terms, blocklist/allowlist is simpler to understand

blacklist and whitelist are common english for so many years, that I think this is none-sense aimed to whitewash (pun somewhat intended) inherently political choices.

Not against either choice, but preferably projects shouldn't be making breaking changes based on political, sociological or marketing preferences. (basically stick with what you have, unless it's *technically* prefered/required to be changed)

*To be clear: when I refer to "political" I'm not pointing fingers which part of the spectrum is better or worse. Just pointing out the fact that inclusive-language focus is very dependent on political affiliation.*

**Comment by dyfrgi** [comment]
It's certainly a political choice to do this renaming, but so are all choices in group projects. Choosing to take no action is also political. Suppose that the original developer had used an anti-Jewish slur to refer to some variable. It would obviously be political to choose not to change that.

And so the same is true here. The offense is lesser and the work is greater than a rename of an internal variable, but those are part of the general calculus of prioritization and not aspects which make it political as opposed to not. Similarly, there are fewer people who would argue that an anti-Jewish slur is an acceptable name for a variable in a program like this one than who argue that whitelist/blacklist is acceptable language, but that just makes it a smaller group, not apolitical.

You can make apolitical choices when you have no communication with other humans and use no shared resources. Until then, it's impossible.

For this specific change, it looks like backward-compatibility is a requirement for monitoring and configuration. Are there other backward-compatibility concerns?

**Comment by Ornias1993** [comment]
> Choosing to take no action is also political.

prime part of my issue with this is the breaking changes for political reasons. 
Its a bad form to have sonething non-technical, influence making inherently bad (=breaking) changes.

on the other way around: I would be 100% okey with doing this change when the code structure already was getting a breaking change anyway.

tldr: it should not be done JUST for political changes, only if there was a breaking change to these varuables anyway.

**Comment by ThinkChaos** [comment]
> Its a bad form to have sonething non-technical, influence making inherently bad (=breaking) changes.

You've been repeating that as if it's a fact. It's not, it's subjective and each project should make their own choices.

I think there's nothing holding us back from renaming the options, and keeping the old ones for a couple releases to help with the transition, just like we do with other options.

**Comment by erichard** [comment]
> blacklist and whitelist are common english for so many years

So was the N word and many more offending terms, the fact they are commons does not make then less offending for certain people. 

Beside of that English is not the native language for many computer scientists or hobbyist, so common or not, the actual semantic of black/white lists is broken. Try to explain to a child why a blacklist is a list of bad things.

> I think there's nothing holding us back from renaming the options, and keeping the old ones for a couple releases to help with the transition, just like we do with other options.

If this is possible then changes can be made without BC break so there is littles (no) reasons to not make the changes. We only need some good will to actually do to work. The #349 seems a good start but it's already 1 year old so it may require some catchups.

**Comment by Ornias1993** [comment]
> > Its a bad form to have sonething non-technical, influence making inherently bad (=breaking) changes.
> 
> You've been repeating that as if it's a fact. It's not, it's subjective and each project should make their own choices.

Correct, but making breaking changes without technical reasons, is called "unreliable". Being an unreliable project == bad in my book.

**Comment by Ornias1993** [comment]
> Between a breaking change and no change at all I proposed a transistioning time in #349 .
> 
> > Maybe we could duplicate problematic endpoints and configurations and mark the old ones as deprecated?
> 
> > Users would get a transition time of some releases to change there setup and the deprecated endpoints could be removed later.
> 
> 
> 
> For the transisting time conversion of configuration values would be commented with a warning to use the new field.  
> 
> Additionally the documentation would only reference the new options and endpoints.
> 
> 
> 
> Support for the old options and endpoints could be scrapped a few versions later or if there are other breaking changes.

As long as breaking changes are bundled and properly semver versioned we should push forward with this!

**Comment by ThinkChaos** [comment]
Since the last comments here, we've added a config migration system so we can do the change and keep backwards-compatibility more easily.
Also see [RFC-8499 - DNS Terminology](https://www.rfc-editor.org/rfc/rfc8499) for a related update to the official standards.

I'll take a look at actually doing this soon :)
